cmd/renterd/main.go

```diff
@@ -116,40 +134,48 @@ func main() {
 		sub: make(map[string]treeMux),
 	}
 
-	if busCfg.enabled {
-		b, cleanup, err := newBus(busCfg, *dir, getWalletKey())
+	var wb worker.Bus
+	var ab autopilot.Bus
+	if busCfg.remoteAddr != "" {
+		b := bus.NewClient(busCfg.remoteAddr, busCfg.apiPassword)
+		wb = b
+		ab = b
+	} else {
+		b, cleanup, err := node.NewBus(busCfg.BusConfig, *dir, getWalletKey())
 		if err != nil {
 			log.Fatal(err)
 		}
 		defer cleanup()
 		log.Println("bus: Listening on", b.GatewayAddress())
-		mux.sub["/api/store"] = treeMux{h: auth(bus.NewServer(b))}
-		autopilotCfg.busAddr = *apiAddr + "/bus/"
-		autopilotCfg.busPassword = getAPIPassword()
+		mux.sub["/api/bus"] = treeMux{h: auth(bus.NewServer(b))}
+		panic("TODO: unify bus interfaces")
+		// wb = b
+		// ab = b
```

Comment:
I reworked the initialization a bit and this is what I came up with. This allows us to initialize the worker and autopilot with either a local bus or a remote Client.

Of course, in order for this to work, those two things have to actually implement the same interface. As it stands, there are a *lot* of discrepancies between the client types and the objects themselves. We'll have to do a big pass to clean all of that up. 😬